dcpy/lifecycle/ingest/transform.py

```diff
@@ -248,18 +247,21 @@ def validate_processing_steps(
     return compiled_steps
 
 
-def run_processing_steps(
-    dataset_name: str,
-    processing_steps: list[FunctionCall],
+def preprocess(
+    config: Config.Preprocessing,
     input_path: Path,
-    output_path,
+    output_path: Path,
 ):
-    """Validates and runs preprocessing steps defined in config object"""
-    compiled_steps = validate_processing_steps(dataset_name, processing_steps)
-    if len(compiled_steps) == 0:
-        shutil.copy(input_path, output_path)
-    else:
-        df = pd.read_parquet(input_path)
-        for step in compiled_steps:
-            df = step(df)
-        df.to_parquet(output_path)
+    """Reprojects if needed, then validates and runs preprocessing steps defined in config object"""
+    df = data.read_parquet(input_path)
+    if config.target_crs:
```

Comment:
Should this just be a processing step? (ie should `validate_processing_steps` just tack this on at the beginning of the `processing_steps`?)

Reply:
Yeah maybe. It definitely shouldn't be in the templates, but it probably makes sense to use the same machinery to run it

Reply:
See [here](https://github.com/NYCPlanning/data-engineering/pull/857/commits/9cb04d05b1c5c69b1b7d782a3228f540de405e5b). Basically, when converting from template to config, if target_crs is specified this is added as the first processing step. Could be argument to put it last. Anyways, this makes things a little cleaner imo - `transform` doesn't need to treat this as a special case now